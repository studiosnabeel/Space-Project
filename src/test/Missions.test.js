import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import MissionsList from '../Components/Missions/missionsList';
import store from '../redux/configureStore';

describe('Missions loads', () => {
  test('the component renders', () => {
    render(
      <Provider store={store}>
        <MissionsList />
      </Provider>,
    );
    expect(screen.getByText('Missions')).toBeInTheDocument();
  });
});
