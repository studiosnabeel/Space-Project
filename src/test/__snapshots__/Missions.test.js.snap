// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Missions component Should render Missions component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <tr
        class="bg-gray-100 border-b even:bg-gray-50"
      >
        <td
          class="px-6 py-4 whitespace-nowrap text-sm font-medium text-gray-900"
        />
        <td
          class="text-sm text-gray-900 font-light px-6 py-4 whitespace-wrap"
        />
        <td
          class="text-sm text-gray-900 font-light px-6 py-4 whitespace-nowrap"
        >
          <p
            class="bg-neutral-700 text-white rounded w-fit px-1"
          >
            Not a member
          </p>
        </td>
        <td
          class="text-sm text-gray-900 font-light px-6 py-4 whitespace-nowrap"
        >
          <button
            class="border-solid rounded border-2 border-[#4b5056] px-4 py-1"
            type="button"
          >
            Join Mission
          </button>
        </td>
      </tr>
    </div>
  </body>,
  "container": <div>
    <tr
      class="bg-gray-100 border-b even:bg-gray-50"
    >
      <td
        class="px-6 py-4 whitespace-nowrap text-sm font-medium text-gray-900"
      />
      <td
        class="text-sm text-gray-900 font-light px-6 py-4 whitespace-wrap"
      />
      <td
        class="text-sm text-gray-900 font-light px-6 py-4 whitespace-nowrap"
      >
        <p
          class="bg-neutral-700 text-white rounded w-fit px-1"
        >
          Not a member
        </p>
      </td>
      <td
        class="text-sm text-gray-900 font-light px-6 py-4 whitespace-nowrap"
      >
        <button
          class="border-solid rounded border-2 border-[#4b5056] px-4 py-1"
          type="button"
        >
          Join Mission
        </button>
      </td>
    </tr>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
